src/app/services/resource.service.ts

```diff
@@ -259,4 +261,36 @@ export class ResourceService {
       });
   }
 
+  /**
+   * Gets all elements from the study dimension that satisfy the constaint if given
+   * @param {Constraint} constraint
+   * @returns {Observable<string[]>}
+   */
+  getStudyNames(constraint: Constraint): Observable<string[]> {
+    return this.transmartResourceService.getStudyNames(constraint)
+      .map((elements: TransmartStudyDimensionElements[]) => {
```

Comment:
dont we already have the resourceService.getStudies() call for this?

Reply:
resourceService.getStudies() returns all the studies

Reply:
then shouldn't we modify the existing getStudies call so that it takes  constraint as input and outputs the corresponding Study[] ?

Reply:
to get Study[] based on constraint there are 2 back-end calls needed. First `v2/dimensions/study/elements` to get just the study names based on constraints and than `studies/studyId/${studyNames}` to get Study objects. getStudies() is used only for auto-complete, constraints are not needed there, right (than it can use only one back-end call)?

Reply:
just one more question, in future, there will be filtering option on a dimension, e.g. the user may indicate that for the study dimension, he only wants to see study-A, and study-B, but not study-C, so the glowing bear should be able to tell transmart to return a customized table based on this filtering, by sending an API call that includes the filtering study names, right? Are the study names in fact study IDs that can be used to make the API call? If they are in fact IDs, then are they really readable by users?